### PR TITLE
support sharding llama2-70b on more than 8 GPUs

### PR DESCRIPTION
**Pull request by zhuohan123**
This PR is #1122 with some style changes.

cc @wejoncy

### CODE REVIEW

**Comment by WoosukKwon** [comment]
@zhuohan123 A quick question: How did you test this PR?

File: vllm/model_executor/models/llama.py [hunk 1]
```diff
@@ -363,9 +376,13 @@ def load_weights(self,
                     shard_size //= self.quant_config.pack_factor
                     offset //= self.quant_config.pack_factor
 
-                loaded_weight = loaded_weight[
-                    shard_size * tensor_model_parallel_rank:shard_size *
-                    (tensor_model_parallel_rank + 1)]
+                if weight_name != "q_proj":
```
Comment (WoosukKwon) [inline vllm/model_executor/models/llama.py:379]: style nit: what about this?
```suggestion
                if weight_name in ["k_proj", "v_proj"]:
```

**Review by WoosukKwon — COMMENTED** [review]
LGTM! Please test the code out before merging the PR.

BTW. can we add this feature to other models when we refactor TP? I think this can be quite useful.

**Review by WoosukKwon — APPROVED** [review]

**Comment by wejoncy** [comment]
> @zhuohan123 A quick question: How did you test this PR?

Good question.

I test it with 16xV100 GPU in multi-nodes but it seems hard to construct a UT for it.

**Comment by zhuohan123** [comment]
> @zhuohan123 A quick question: How did you test this PR?

I tested 7B LLaMA with single GPU and 2 GPUs. It's impossible for me to test for more than 8 GPUs as well. But the code itself looks good. I will try to test again once we get more hardware.

**Pull request merged**